### DISCUSSION TITLE
Xiaomi S1 : problem with ANALOG display

### DISCUSSION THREAD

**Discussion by palomo232323** [ooflet/Mi-Create#51] [Questions-EN]
I am trying to create a watchface for my Xiaomi S1.

I’m having trouble getting the ANALOG display to work.

I tried decompiling some working .bin files.
I also tried creating a watchface from scratch with MI CREATE, compiling it (and so far everything seems fine), then renaming the output .face file to .bin and decompiling it (just to verify that the file was compiled correctly ...and avoid to brick my watch).

In both cases, if I open the decompiled project with MI CREATE (the .fprj project file created by the decompiler), error messages appear only on the analog display component.
What am I doing wrong?

I also have another question: to upload a watch face created with MI CREATE, if I take the .face file and rename it to .bin, can I upload it directly to the phone (using the Notify for Xiaomi tool, which allows uploading .bin files)?
Or do I necessarily have to publish the .bin file onto m0tral's servers?

**Comment by vonfritz** [comment]
If you don't specify what the error messages are How do we know what the problem is. As for loading the dials, the .face or .bin extension makes no difference, they are practically the same thing. You can do without renaming and directly load the file with the .face extension. I have an S3 and load the dials with notify without problems. You just have to pay attention to the ID which for the dials compiled with the m0tral compiler is always the same, this will not allow you to load more than one dial. Loading a second one with the same ID will overwrite the one loaded previously.

**Reply by palomo232323** [reply]
Hi, thanks for your reply.

Sure, this is the error message:

```
Cannot render project! Unable to create object analogClock_01:
Traceback (most recent call last):
File "C:\xxxxxx\DOWNLO~1\Mi Create\widgets\canvas.py", line 440, in createWidgetFromData 
File "C:\xxxxxx\DOWNLO~1\Mi Create\widgets\canvas.py", line 331, in createAnalogDisplay  
File "C:\xxxxxx\DOWNLO~1\Mi Create\ntpath.py", line 149, in join  
File "C:\xxxxxx\DOWNLO~1\Mi Create\genericpath.py", line 164, in _check_arg_types  
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'NoneType'
```



This message appear when I try to open a project "unpacked" from a .bin file (created with and exported by the Notify for Xiaomi tool).
Same error if a create a new face (with analog display), unpack it and try to open the unpacked files.


This the the fprj content:

```
<?xml version="1.0" encoding="utf-16"?>
<FaceProject DeviceType="3">
<Screen Title="Notify562" Bitmap="preview.png">
<Widget p3:type="FaceWidgetImage" Shape="30" Name="image_00" X="0" Y="0" Width="466" Height="466" Alpha="255" Visible_Src="0" Bitmap="img_0000.png" xmlns:p3="http://www.w3.org/2001/XMLSchema-instance" />
<Widget p3:type="FaceWidgetAnalogClock" Shape="27" Name="analogClock_01" X="0" Y="0" Width="466" Height="466" Alpha="255" Visible_Src="0" HourHandCorrection_En="0" MinuteHandCorrection_En="0" BgImage_rotate_xc="0" BgImage_rotate_yc="0" HourHand_ImageName="img_0001.png" HourImage_rotate_xc="18" HourImage_rotate_yc="227" MinuteHand_Image="img_0002.png" MinuteImage_rotate_xc="18" MinuteImage_rotate_yc="227" SecondHand_Image="img_0003.png" SecondImage_rotate_xc="18" SecondImage_rotate_yc="227" xmlns:p3="http://www.w3.org/2001/XMLSchema-instance" />
</Screen>
</FaceProject>
```



PS:  how can I upload the bin/face file on my watch?  Only publishing on m0tral's servers or can I use the Notify for Xiaomi tool (directly from my smartphone)?

Thanks.

**Reply by ooflet** [reply]
This is an issue with Mi Create that has been fixed on main branch where missing image fields aren't accounted for, you can either run Mi Create from source on either the main or GMF branch (on the GMF branch there are some new features) or just simply add the missing property Background_ImageName="" into the analogClock_01 widget.

You can use the Notify for Xiaomi tool on your smartphone if you want to test out your watchface quickly or just want to use it personally, but if you decide to want to share your watchface with others you can use m0tral's servers to publish your watchface.

**Reply by palomo232323** [reply]
Ok, thanks.... I see, the "unpack" removes the "Background_ImageName" property in the config file.  

Just for my info, to use the fixed version (main or GMF branch), should I compile the source code, or a compiled windows version can be downloaded?

Thanks for the support.

**Reply by ooflet** [reply]
You dont need to compile anything because it's just a Python program, follow instructions on the README to run the program. There are many issues with the latest 4.18 compiler which prevent me from releasing a compiled Windows build or merging the branch until a workaround is found. Unfortunately both Linux and Windows is plagued with issues when not running the compiler on EasyFace.

Looks like bundling both 4.16 and 4.18 is the best option for now before workarounds are found.

**Reply by palomo232323** [reply]
ok, thanks.  I'll try to install Python and follow the instructions...

Just another question: as you wrote, every time a create a new face, the ID is always the same (when I import the bin using Notify for Xiaomi).  Is there some workaround to force/generate different IDs?

Thanks.

**Reply by vonfritz** [reply]
> ok, thanks. I'll try to install Python and follow the instructions...
> 
> Just another question: as you wrote, every time a create a new face, the ID is always the same (when I import the bin using Notify for Xiaomi). Is there some workaround to force/generate different IDs?
> 
> Thanks.
Use Mi-Create from the GMF branch which has the ability to handling watchface ID.
![WFID2](https://github.com/user-attachments/assets/bb8e7cb4-b967-4ec9-bcf5-312f040ae085)
